ocaml/xapi/xapi_host.ml

```diff
@@ -3581,3 +3585,34 @@ let list_timezones ~__context ~self:_ =
     Helpers.call_script !Xapi_globs.timedatectl ["list-timezones"]
     |> Astring.String.cuts ~empty:false ~sep:"\n"
   with e -> Helpers.internal_error "%s" (ExnHelper.string_of_exn e)
+
+let get_ntp_synchronized ~__context ~self:_ =
+  match Xapi_host_ntp.is_synchronized () with
+  | Ok r ->
+      r
+  | Error msg ->
+      Helpers.internal_error "%s" msg
+
+let set_servertime ~__context ~self ~value =
+  let f = Helpers.call_script !Xapi_globs.timedatectl in
+  with_lock time_m @@ fun () ->
+  let tz = Db.Host.get_timezone ~__context ~self in
+  let not_utc = tz <> "UTC" in
+  try
+    (* The [value] stores only a naive date/time and a fixed timezone offset.
+       Convert it to UTC to avoid ambiguities caused by additional timezone
+       details, such as Daylight Saving Time (DST) rules or historical changes. *)
+    match Date.to_utc value with
+    | Some {t; _} ->
```

Comment:
I wouldn't want users to have access to the record. That can be avoided quite easily:

```suggestion
    match Date.to_utc value |> Option.map Date.to_ptime with
    | Some t ->
```

Reply:
Thanks. I missed the existing function `Date.to_ptime`.
Fixed.